jni/src/nmslib_wrapper.cpp

```diff
@@ -243,14 +247,24 @@ jobjectArray knn_jni::nmslib_wrapper::QueryIndex(knn_jni::JNIUtilInterface * jni
         jniUtil->ReleaseFloatArrayElements(env, queryVectorJ, rawQueryvector, JNI_ABORT);
         throw;
     }
+
     jniUtil->ReleaseFloatArrayElements(env, queryVectorJ, rawQueryvector, JNI_ABORT);
+    std::unordered_map<std::string, jobject> methodParams;
+    if (methodParamsJ != nullptr) {
+        methodParams = jniUtil->ConvertJavaMapToCppMap(env, methodParamsJ);
+    }
 
-    similarity::KNNQuery<float> knnQuery(*(indexWrapper->space), queryObject.get(), kJ);
-    indexWrapper->index->Search(&knnQuery);
+    int queryEfSearch = knn_jni::commons::getIntegerMethodParameter(env, jniUtil, methodParams, EF_SEARCH, -1);
+    similarity::KNNQuery<float>* query;
+    if (queryEfSearch == -1) {
+        query = new similarity::KNNQuery<float>(*(indexWrapper->space), queryObject.get(), kJ);
+    } else {
+        query = new similarity::HNSWQuery<float>(*(indexWrapper->space), queryObject.get(), kJ, queryEfSearch);
```

Comment:
Dont we need eps? https://github.com/opensearch-project/k-NN/blob/9db146baba5804f3a49afdf3eb2c3de3ac9d1cb7/jni/patches/nmslib/0002-Adds-ability-to-pass-ef-parameter-in-the-query-for-h.patch#L121

Reply:
There is a default present and we didn't seem to be using the defaults for KNNQuery, so kept the same behavior for hnsw. Can you elaborate why do you think we need it?

Reply:
I think @jmazanec15 missed this line where default value is defined. 
https://github.com/opensearch-project/k-NN/blob/9db146baba5804f3a49afdf3eb2c3de3ac9d1cb7/jni/patches/nmslib/0002-Adds-ability-to-pass-ef-parameter-in-the-query-for-h.patch#L50

Reply:
I think @jmazanec15 missed this line where default is defined?
https://github.com/opensearch-project/k-NN/blob/9db146baba5804f3a49afdf3eb2c3de3ac9d1cb7/jni/patches/nmslib/0002-Adds-ability-to-pass-ef-parameter-in-the-query-for-h.patch#L50

Reply:
Will this be using the HNSWQuery constructor? Or will it fall through to the parent class and somehow cast ef to eps? My concern here is that this signature doesnt look like it would support a default parameter for eps in c++:
```
HNSWQuery<dist_t>::HNSWQuery(const Space<dist_t> &space, const Object* query_object, const unsigned K, unsigned ef, float eps) 
```
I would expect defaults to be like `float eps = 0.0f`.

Reply:
Im not sure how for HnswQuery(, the eps parameter can be inferred as optional based on superclass.

Reply:
Oh I see you defaulted here: https://github.com/opensearch-project/k-NN/blob/9db146baba5804f3a49afdf3eb2c3de3ac9d1cb7/jni/patches/nmslib/0002-Adds-ability-to-pass-ef-parameter-in-the-query-for-h.patch#L50

Reply:
Thats good then.

Why did we set ef = 10 by default?

Reply:
I was attempting to integrate with the integration test in nmslib. I failed since its way too rigid to add additional parameters. As a part of it I had to give a default and gave it 10 and then kept it. I can change it if needed

Reply:
Will it ever fall through to default? If not, thats okay. If yes, then we should sync it with our current defaults (100)

Reply:
It won't as of now but I will change it 100 to be defensive wrt future code changes